datalad/conftest.py

```diff
@@ -126,8 +127,17 @@ def prep_tmphome():
                 new_home, _ = prep_tmphome()
                 m.enter_context(patch.dict(os.environ, get_home_envvars(new_home)))
         else:
-            _, cfg_file = prep_tmphome()
-            m.enter_context(patch.dict(os.environ, {'GIT_CONFIG_GLOBAL': str(cfg_file)}))
+            new_home, cfg_file = prep_tmphome()
+            if 'GIT_HOME' in os.environ:
+                home_env = {'HOME': os.environ['GIT_HOME']}
+            else:
+                home_env = get_home_envvars(new_home)
+            # Some software does not expect a non-standard HOME
+            # point GIT_CONFIG_GLOBAL to the new git user config
+            # specifically to mitigate that
+            # see c0fab80374bc3e128453dd1bd5086d17c8e1186f
+            home_env['GIT_CONFIG_GLOBAL'] = str(cfg_file)
+            m.enter_context(patch.dict(os.environ, home_env))
```

Comment:
seems becomes at large duplication with code above...
can't do suggestion over unchanged codeblock above, so - 
Overall -- if to go this way, can't we just `if` on git for `GIT_CONFIG_GLOBAL` only? i.e. 

```python
            new_home, cfg_file = prep_tmphome()
            if 'GIT_HOME' in os.environ:
                home_env = {'HOME': os.environ['GIT_HOME']}
            else:
                home_env = get_home_envvars(new_home)
            # Some software does not expect a non-standard HOME
            # point GIT_CONFIG_GLOBAL to the new git user config
            # specifically to mitigate that
            # see c0fab80374bc3e128453dd1bd5086d17c8e1186f
            if external_versions['cmd:git'] >= "2.32":
                 home_env['GIT_CONFIG_GLOBAL'] = str(cfg_file)
            m.enter_context(patch.dict(os.environ, home_env))
```

but I am afraid that this would bring us back what was attempted to be fixed by #6260 et al

Reply:
> but I am afraid that this would bring us back what was attempted to be fixed by https://github.com/datalad/datalad/pull/6260 et al

yes indeed, the change you propose above would cause ``USERPROFILE`` to be overwritten on any Windows machine - it would probably cause quite a few tests to fail, and add a major obfuscation to debugging.